### PR TITLE
Add --show-splash command line option to control whether splash displays

### PR DESCRIPTION
**Pull request by knitzsche**
Default is to display splash. --show-splash=false suppresses it. (Fixes ~ #830)

### CODE REVIEW

File: examples/example-server-lib/sw_splash.cpp [hunk 1]
```diff
@@ -143,6 +142,8 @@ struct SwSplash::Self : SplashSession
 
     Globals globals;
 
+    bool splash;
+
```
Comment (AlanGriffiths) [inline examples/example-server-lib/sw_splash.cpp:146]: Probably better to avoid default initialize for a bool. Vis:
```C++
    bool splash = true;
```
And maybe better to call it `show_splash`.

File: examples/example-server-lib/sw_splash.cpp [hunk 2]
```diff
@@ -174,6 +179,8 @@ SwSplash::operator std::shared_ptr<SplashSession>() const
 
 void SwSplash::Self::operator()(struct wl_display* display)
 {
+    if (!splash)
+        return;
```
Comment (AlanGriffiths) [inline examples/example-server-lib/sw_splash.cpp:183]: This would be clearer as:
```C++
    if (!show_splash)
        return;
```

File: examples/example-server-lib/sw_splash.h [hunk 3]
```diff
@@ -28,6 +28,7 @@ class SwSplash
     SwSplash();
     ~SwSplash();
 
+    void operator()(bool splash_opt);
```
Comment (AlanGriffiths) [inline examples/example-server-lib/sw_splash.h:31]: I know other functions on this are done as `operator()`s, but this is probably better as a normal member function. E.g. `void enable(bool show_splash)`

File: examples/miral-kiosk/kiosk_main.cpp [hunk 4]
```diff
@@ -108,6 +109,12 @@ int main(int argc, char const* argv[])
     MirRunner runner{argc, argv};
 
     DisplayConfiguration display_config{runner};
+ 
+    CommandLineOption show_splash{
+        [&](bool show_splash) {splash(show_splash); },
+        "splash",
+        "Control display splash on startup",
```
Comment (AlanGriffiths) [inline examples/miral-kiosk/kiosk_main.cpp:116]: The option name (and explanation) leaves a little to be desired, I would prefer "show-splash".

File: examples/miral-kiosk/kiosk_main.cpp [hunk 5]
```diff
@@ -79,6 +79,7 @@ struct KioskAuthorizer : miral::ApplicationAuthorizer
 };
 
 std::atomic<bool> KioskAuthorizer::startup_only{false};
+
```
Comment (AlanGriffiths) [inline examples/miral-kiosk/kiosk_main.cpp:82]: Unrelated?

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Thanks for this, it does what it says. However,  few renames would make it clearer.

**Review by RAOF — COMMENTED** [review]
It's a *little* weird that this disables the splash client by making the client start up and then not do anything rather than just not starting up the splash client, but I can see how this is an easier change to make.

**Comment by AlanGriffiths** [comment]
> It's a _little_ weird that this disables the splash client by making the client start up and then not do anything rather than just not starting up the splash client, but I can see how this is an easier change to make.

That wouldn't be too hard. It just needs something similar to egmde's `run_apps` lambda (but using `InternalClientLauncher`). See: https://github.com/AlanGriffiths/egmde/blob/master/egmde.cpp#L52

File: examples/example-server-lib/sw_splash.cpp (hunk 1 above)
Comment (knitzsche) [reply examples/example-server-lib/sw_splash.cpp]: fixed in pr update

File: examples/example-server-lib/sw_splash.cpp (hunk 2 above)
Comment (knitzsche) [reply examples/example-server-lib/sw_splash.cpp]: fixed in pr update

File: examples/example-server-lib/sw_splash.h (hunk 3 above)
Comment (knitzsche) [reply examples/example-server-lib/sw_splash.h]: fixed in pr update

File: examples/miral-kiosk/kiosk_main.cpp (hunk 4 above)
Comment (knitzsche) [reply examples/miral-kiosk/kiosk_main.cpp]: fixed in pr update

File: examples/miral-kiosk/kiosk_main.cpp (hunk 5 above)
Comment (knitzsche) [reply examples/miral-kiosk/kiosk_main.cpp]: removed accidental extra line in pr update

**Comment by knitzsche** [comment]
Alan, I made the changes you requested and updated the PR branch, but the PR still shows "changes requested". Do I need to do something to update this?

**Comment by AlanGriffiths** [comment]
> Do I need to do something to update this?

No, I just have to get around to re-reviewing. (It's on my list.)

**Review by AlanGriffiths — APPROVED** [review]
LGTM. I'll leave @RAOF have a final say.

**Comment by knitzsche** [comment]
Hi folks. Just checking in since this has an impact on a project. Cheers.

**Review by wmww — APPROVED** [review]
LGTM
bors r+
